csharp/src/elements/Geometry/Polygon.cs

```diff
@@ -91,19 +132,163 @@ public bool Contains(Polygon polygon)
+        /// </returns>
+        public bool Contains(IList<Polygon> polygons)
+        {
+            var thisPath = this.ToClipperPath();
+            foreach (Polygon polygon in polygons)
+            {
+                var polyPath = polygon.ToClipperPath();
+                foreach (IntPoint vertex in polyPath)
+                {
+                    if (Clipper.PointInPolygon(vertex, thisPath) != 1)
+                    {
+                        return false;
+                    }
+                }
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Tests if the supplied 2D Vector3 point is within or touches the perimeter of this Polygon.
+        /// </summary>
+        /// <param name="point">The 2D Vector3 point to compare.</param>
         /// <returns>
-        /// Returns true if every vertex of the supplied Polygon falls within or on the perimeter of this Polygon.
+        /// Returns true if the supplied 2D Vector3 point falls within or touches the perimeter of this Polygon.
+        /// </returns>
+        public bool Covers(Vector3 point)
+        {
+            var scale = 1024.0;
+            var thisPath = this.ToClipperPath();
+            var intPoint = new IntPoint(point.X * scale, point.Y * scale);
+            if (Clipper.PointInPolygon(intPoint, thisPath) == 0)
+            {
+                return false;
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Tests if all the 2D Vector3 points in the supplied list fall within or on the perimeter of this Polygon.
+        /// </summary>
+        /// <param name="points">The list of 2D Vector3 points to compare.</param>
+        /// <returns>
+        /// Returns true if any of the supplied 2D Vector3 points fall within or on the perimeter of this Polygon.
+        /// </returns>
+        public bool Covers(IList<Vector3> points)
+        {
+            var scale = 1024.0;
+            var thisPath = this.ToClipperPath();
+            foreach (Vector3 point in points)
+            {
+                var intPoint = new IntPoint(point.X * scale, point.Y * scale);
+                if (Clipper.PointInPolygon(intPoint, thisPath) == 0)
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Tests if this Polygon covers the supplied Polygon. This Polygon covers the supplied Polygon if all points of the supplied Polygon fall within or on the perimeter of this Polygon.
+        /// </summary>
+        /// <param name="polygon">The Polygon to test for coverage by this Polygon.</param>
+        /// <returns>
+        /// Returns true if the supplied Polygon is contained by or coincides with the perimeter of the supplied Polygon.
         /// </returns>
         public bool Covers(Polygon polygon)
         {
+            var tolerance = 0.00001;
```

Comment:
This is a magic number. If we always use this tolerance, it should be a `const` at the top of the `Polygon` class.

Reply:
Yup it's magic. Should we do the same with the 1024.0 scaling factor needed by Clipper?

Reply:
Done for both tolerance (now "areaTolerance" and scale)